klient/k8s/res/resource.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package res
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+)
+
+type Resources struct {
+	// config is the rest.Config to talk to an apiserver
+	config *rest.Config
+
+	// scheme will be used to map go structs to GroupVersionKinds
+	scheme *runtime.Scheme
+}
+
+func Res(cfg *rest.Config) *Resources {
+	if cfg == nil {
+		fmt.Println("must provide rest.Config")
+		return nil
+	}
+
+	res := &Resources{
+		config: cfg,
+		scheme: scheme.Scheme,
+	}
+
+	return res
+}
+
+// New wrapper fumction for controller runtimr client New()
+func (r *Resources) New() cr.Client {
```

Comment:
This method is leaking a controller-runtime client. is that right ? Probably should think about wrapping it into a klient.Client or something.

Reply:
Actually, better yet, dont make make `New()` visible because all interaction should go through the `Res()` constructor function any way.

Reply:
As I mentioned, let's create and store the cr.Client in the Resource struct. Then all this goes away.